internal/shell/shell.go

```diff
@@ -578,6 +581,9 @@ func (s *Shell) executeCommand(ctx context.Context, cmdCfg process.Config, stdou
 			defer stdErrStreamer.Close()
 			cmdCfg.Stderr = stdErrStreamer
 		}
+
+		// This should respect the log format we set for the agent
+		processLogger = logger.NewConsoleLogger(logger.NewTextPrinter(os.Stderr), os.Exit)
```

Comment:
Actually there is a `Logger` embedded in `Shell`, so you can do this:

```suggestion
		processLogger = s.Logger
```

Reply:
This doesn't work because `s.Logger` is a `shell.Logger` interface not a `logger.Logger` interface, which are completely different. Why do we have multiple logger interfaces?

Reply:
Oh what? argh

Reply:
At least don't write to `os.Stderr` - that would bypass the secret redactor. Use `cmdCfg.Stderr` I guess?